### ISSUE TITLE
How to Finetune  and Semi-supervised evaluate AimCLR?

### ISSUE THREAD

**Issue by yanghhx** [Levigty/AimCLR#3]
I am very interested in your work and thanks for your sharing. I'd like to know how to finetune and semi-supervised evaluate AimCLR? Have the two evaluation been included in your code? If not , could you please share them for me? My email is :qinyangz@163.com

**Comment by Levigty** [comment]
Thank you for your interest in our work. Linear evaluation fixes the encoder with [“linear_evaluation.py”](https://github.com/Levigty/AimCLR/blob/main/processor/linear_evaluation.py) line 40-53. Finetune and semi-supervised evaluation just do not fix the encoder. 
You can also use the trained weights as the initial weights of [ST-GCN](https://github.com/yysijie/st-gcn/blob/master/OLD_README.md) and ignore some weights to do finetune/semi-supervised evaluation. Semi-supervised dataloader has already in [ntu_feeder.py](https://github.com/Levigty/AimCLR/blob/main/feeder/ntu_feeder.py).

**Comment by yanghhx** [comment]
Thanks for your answer. However,  when I run the orignina code which is set the "param.requires_grad = False" in line 40-53, the accuracy is only about 46.5%. When I set the "param.requires_grad =True" in line 40-53 of linear_evaluation.py, the linear evalution accuracy of NTU60 view can reach 83.8%. However , the operation may be the fine-tuning operation that is described in your paper. Can you please tell me the reason?
















At 2022-03-27 12:55:42, "Gty" ***@***.***> wrote:

Thank you for your interest in our work. Linear evaluation fixes the encoder with “linear_evaluation.py” line 40-53. Finetune and semi-supervised evaluation just do not fix the encoder.
You can also use the trained weights as the initial weights of ST-GCN and ignore some weights to do finetune/semi-supervised evaluation. Semi-supervised dataloader has already in ntu_feeder.py.

**Comment by Levigty** [comment]
I have released the code of finetune in `processor`, you can check it. For finetune, the learning rate of 0.01 or 0.001 may be the better choices.

**Comment by yanghhx** [comment]
Thanks for your sharing.  I am very intersted in your work. Do you have your web where your work is relased? Can I pay attention to it?
















At 2022-04-02 10:20:17, "Gty" ***@***.***> wrote:

I have released the code of finetune in processor, you can check it. For finetune, the learning rate of 0.01 or 0.001 may be the better choices.

**Comment by Levigty** [comment]
Thank you for your attention to our work. There is no dedicated web page, but the resources of the paper are available.

**Comment by yanghhx** [comment]
Thanks for your answer. Iam sorry that I have an another question. When I run the origninal linear_evalution  code which is set the "param.requires_grad = False" in line 40-53, the accuracy is only about 46.5%.  
















At 2022-04-02 10:43:29, "Gty" ***@***.***> wrote:

Thank you for your attention to our work. There is no dedicated web page, but the resources of the paper are available.

**Comment by Levigty** [comment]
You should indicate the model you used was trained by yourself or the released, and whether you strictly followed the steps in the `readme` and have you change some code.

**Comment by yanghhx** [comment]
I have strictly followed the steps in the readme to process the ntu60 datasets and train the model by myself with your code, however the accuracy of my model is  46.5% . When I modify the linear_evalution.py to set the "param.requires_grad =True" in line 40-53, the linear evalution accuracy of NTU60 view can reach 83.8%.










At 2022-04-02 10:52:43, "Gty" ***@***.***> wrote:

You should indicate the model you used was trained by yourself or the released, and whether you strictly followed the steps in the readme and have you change some code.
